statics/js/components/graph-layout.js

```diff
@@ -1,6 +1,187 @@
+        return false;
+      }
+
+      return this.bandwidthCheck > this.bandwidth.active;
+    },
+
+    getText: function() {
+      return bandwidthToString(this.bandwidthAbsolute);
+    },
+
+    isActive: function() {
+      return (this.bandwidthCheck > this.bandwidth.active) && (this.bandwidthCheck < this.bandwidth.warning);
+    },
+
+    isWarning: function() {
+      return (this.bandwidthCheck >= this.bandwidth.warning) && (this.bandwidthCheck < this.bandwidth.alert);
+    },
+
+    isAlert: function() {
+      return this.bandwidthCheck >= this.bandwidth.alert;
+    },
+  },
+});
+
+var LinkLabelLatency = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+    setup: function(topology, link) {
+      this.active = 0;
+      this.warning = 10
+      this.alert = 100;
+      this.link = link;
+      this.updateData();                    // prepare data for next cycle
+    },
+
+    flowKey: function(a) {
+      return `TrackingID:${a.TrackingID},ANodeTID:${a.ANodeTID},BNodeTID:${a.BNodeTID}`;
+    },
+
+    linklabel: function(a, b) {
+      return `${a.Name}-->${b.Name}`;
+    },
+
+    flowCompare: function(a, b) {
+      keya = this.flowKey(a);
+      keyb = this.flowKey(b);
+      return keya.localeCompare(keyb);
+    },
+
+    updateLatency: function(flows) {
+      let prev, curr;
+      for (i in flows) {
+        prev = curr;
+        curr = flows[i];
+
+        if (!prev) {
+          continue;
+        }
+
+        if (self.flowKey(curr) != self.flowKey(prev)) {
+          continue;
+        }
+
+        latency = Math.abs(curr.Start - prev.Start);
+        this.link.latencyTimestamp = Math.max(curr.Start, prev.Start);
+        this.link.latency = latency;
+        return true;
+      }
+      return false;
+    },
+
+    updateData: function() {
+      self = this;
+
+      const a = this.link.source.metadata;
+      const b = this.link.target.metadata;
+      const prefix = `${a.Name}<-->${b.Name}`
+
+      if (!a.Capture) {
+        return;
+      }
+      if (!b.Capture) {
+        return;
+      }
+
+      const limit = 10;
+      const anyNode = 'Within("'+a.TID+'", "'+b.TID+'")';
+      const query = 'G.Flows().Has("NodeTID", '+anyNode+').Limit('+limit+')';
+      console.log(`${prefix}: ${query}`)
+
+      this.$topologyQuery(query)
+        .then(function(flows) {
+          if (flows.length === 0) {
+            return;
+          }
+          flows = flows.sort(self.flowCompare);
+          if (self.updateLatency(flows)) {
+            console.log(`${prefix}: ${self.getText()}`);
+          }
+        })
+        .fail(function(error) {
+          console.log(error);
+        });
+    },
+
+    hasData: function() {
+      if (!this.link.latencyTimestamp) {
+        return false;
+      }
+      elapsedMillis = Date.now() - new Date(this.link.latencyTimestamp);
+      return elapsedMillis <= maxClockSkewMillis;
+    },
+
+    getText: function() {
+      return `${this.link.latency} ms`;
```

Comment:
should be divided by 1000000 as you add 'ms'

Reply:
used `Math.abs(curr.RTT-prev.RTT)/1000`